code42_on_poll_connector.py

```diff
@@ -0,0 +1,437 @@
+from datetime import datetime
+
+import dateutil.parser
+import phantom.app as phantom
+from code42_util import get_thirty_days_ago, build_alerts_query
+from py42.sdk.queries.fileevents.file_event_query import FileEventQuery
+from py42.sdk.queries.fileevents.filters import (
+    ExposureType,
+    DeviceUsername,
+    Actor,
+    EventTimestamp,
+    EventType,
+    FileCategory,
+)
+
+"""The contents of this module that related to mapping alert observations to file events borrows heavily from the
+Code42 Cortex XSOAR integration as well as the code42cli python package.
+"""
+
+
+JSON_TO_CEF_MAP = {
+    "actor": "suser",
+    "cloudDriveId": "aid",
+    "createTimestamp": "fileCreateTime",
+    "deviceUid": "deviceExternalId",
+    "deviceUserName": "suser",
+    "domainName": "dvchost",
+    "emailRecipients": "duser",
+    "emailSender": "suser",
+    "eventId": "externalId",
+    "eventTimestamp": "end",
+    "exposure": "reason",
+    "fileCategory": "fileType",
+    "fileName": "fname",
+    "filePath": "filePath",
+    "fileSize": "fsize",
+    "insertionTimestamp": "rt",
+    "md5Checksum": "fileHash",
```

Comment:
Could we also / instead populate the `fileHashMd5` property on the artifact CEF? I'm trying to configure the input to `hunt file` in the playbook to pull the hash from the artifact, but it seems like the `contains` for that parameter only allows the values `fileHashMd5` and `fileHashSha256` from the CEF.

Reply:
Interesting! There seems to be a connection between `contains` and CEF.
Yes, we can just use values from `contains`.